controllers/helpers/create-form-router.js

```diff
@@ -59,11 +71,24 @@ function createFormRouter({ router, formModel }) {
             }
         }
 
+        // for users submitting a step, increase their session expiry
+        // so they can save progress beyond a browser session
+        function extendSessionDuration(req) {
+            req.session.cookie.maxAge = moment()
+                .add(EXTENDED_SESSION_DURATION, 'days')
+                .toDate();
+            req.flash('progressSaved', {
```

Comment:
Don't know if we need to use `flash` here, you could probably assign this to `res.locals.progressSaved` and use it directly in the template at that point, or at least from `renderStep`

Reply:
The challenge there is that `renderStep` is only called if there's a validation error, otherwise they're redirected to the next form step (which obviously loses the `res.locals` data), that's why I used `flash` initially...

Reply:
Ah! missed that. Carry on.